Build failure: fastcap

https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.fastcap.x86_64-linux

## Steps To Reproduce

Steps to reproduce the behavior:

1. `nix build github:nixos/nixpkgs/nixos-unstable#fastcap`


## Build log

<details>

<summary>Build Log</summary>

```
mulGlobal.h:128:23: warning: incompatible implicit declaration of built-in function 'exit' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch8;;]
  128 |        if(FLAG == ON) exit(0);                                              \
      |                       ^~~~
mulSetup.c:827:17: note: in expansion of macro 'CALLOC'
  827 |                 CALLOC(nc->multi, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:128:23: note: include '<stdlib.h>' or provide a declaration of 'exit'
  128 |        if(FLAG == ON) exit(0);                                              \
      |                       ^~~~
mulSetup.c:827:17: note: in expansion of macro 'CALLOC'
  827 |                 CALLOC(nc->multi, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:144:10: warning: incompatible implicit declaration of built-in function 'exit' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch8;;]
  144 |          exit(0);                                                           \
      |          ^~~~
mulSetup.c:827:17: note: in expansion of macro 'CALLOC'
  827 |                 CALLOC(nc->multi, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:144:10: note: include '<stdlib.h>' or provide a declaration of 'exit'
  144 |          exit(0);                                                           \
      |          ^~~~
mulSetup.c:827:17: note: in expansion of macro 'CALLOC'
  827 |                 CALLOC(nc->multi, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:118:22: warning: cast to pointer from integer of different size [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wint-to-pointer-cast-Wint-to-pointer-cast8;;]
  118 |      else if(((PNTR)=(TYPE*)CALCORE(NUM, TYPE))==NULL) {                    \
      |                      ^
mulSetup.c:839:17: note: in expansion of macro 'CALLOC'
  839 |                 CALLOC(nc->local, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:122:30: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long unsigned int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-Wformat=8;;]
  122 |        (void)fprintf(stderr, " (NULL pointer on %d byte request)\n",        \
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  123 |                      (NUM)*sizeof(TYPE));                                   \
      |                      ~~~~~~~~~~~~~~~~~~
      |                           |
      |                           long unsigned int
mulSetup.c:839:17: note: in expansion of macro 'CALLOC'
  839 |                 CALLOC(nc->local, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:122:50: note: format string is defined here
  122 |        (void)fprintf(stderr, " (NULL pointer on %d byte request)\n",        \
      |                                                 ~^
      |                                                  |
      |                                                  int
      |                                                 %ld
mulGlobal.h:108:17: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-Wformat=8;;]
  108 |                 "Total Memory Allocated: %d kilobytes (brk = 0x%x)\n", \
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  109 |                 memcount/1024, sbrk(0));                               \
      |                 ~~~~~~~~~~~~~
      |                         |
      |                         long int
mulGlobal.h:124:8: note: in expansion of macro 'DUMPALLOCSIZ'
  124 |        DUMPALLOCSIZ;                                                        \
      |        ^~~~~~~~~~~~
mulSetup.c:839:17: note: in expansion of macro 'CALLOC'
  839 |                 CALLOC(nc->local, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:108:43: note: format string is defined here
  108 |                 "Total Memory Allocated: %d kilobytes (brk = 0x%x)\n", \
      |                                          ~^
      |                                           |
      |                                           int
      |                                          %ld
mulGlobal.h:128:23: warning: incompatible implicit declaration of built-in function 'exit' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch8;;]
  128 |        if(FLAG == ON) exit(0);                                              \
      |                       ^~~~
mulSetup.c:839:17: note: in expansion of macro 'CALLOC'
  839 |                 CALLOC(nc->local, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:128:23: note: include '<stdlib.h>' or provide a declaration of 'exit'
  128 |        if(FLAG == ON) exit(0);                                              \
      |                       ^~~~
mulSetup.c:839:17: note: in expansion of macro 'CALLOC'
  839 |                 CALLOC(nc->local, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:144:10: warning: incompatible implicit declaration of built-in function 'exit' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch8;;]
  144 |          exit(0);                                                           \
      |          ^~~~
mulSetup.c:839:17: note: in expansion of macro 'CALLOC'
  839 |                 CALLOC(nc->local, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:144:10: note: include '<stdlib.h>' or provide a declaration of 'exit'
  144 |          exit(0);                                                           \
      |          ^~~~
mulSetup.c:839:17: note: in expansion of macro 'CALLOC'
  839 |                 CALLOC(nc->local, numterms, double, ON, AMSC);
      |                 ^~~~~~
mulSetup.c: At top level:
mulSetup.c:859:8: error: return type defaults to 'int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int8;;]
  859 | static setMaxq(sys)
      |        ^~~~~~~
mulSetup.c:952:8: error: return type defaults to 'int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int8;;]
  952 | static markUp(child, flag)
      |        ^~~~~~
mulSetup.c:970:8: error: return type defaults to 'int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int8;;]
  970 | static getInter(child)
      |        ^~~~~~~~
mulSetup.c: In function 'getInter':
mulGlobal.h:118:22: warning: cast to pointer from integer of different size [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wint-to-pointer-cast-Wint-to-pointer-cast8;;]
  118 |      else if(((PNTR)=(TYPE*)CALCORE(NUM, TYPE))==NULL) {                    \
      |                      ^
mulSetup.c:1031:17: note: in expansion of macro 'CALLOC'
 1031 |   if(vects > 0) CALLOC(child->interList, vects, cube*, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:122:30: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long unsigned int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-Wformat=8;;]
  122 |        (void)fprintf(stderr, " (NULL pointer on %d byte request)\n",        \
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  123 |                      (NUM)*sizeof(TYPE));                                   \
      |                      ~~~~~~~~~~~~~~~~~~
      |                           |
      |                           long unsigned int
mulSetup.c:1031:17: note: in expansion of macro 'CALLOC'
 1031 |   if(vects > 0) CALLOC(child->interList, vects, cube*, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:122:50: note: format string is defined here
  122 |        (void)fprintf(stderr, " (NULL pointer on %d byte request)\n",        \
      |                                                 ~^
      |                                                  |
      |                                                  int
      |                                                 %ld
mulGlobal.h:108:17: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-Wformat=8;;]
  108 |                 "Total Memory Allocated: %d kilobytes (brk = 0x%x)\n", \
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  109 |                 memcount/1024, sbrk(0));                               \
      |                 ~~~~~~~~~~~~~
      |                         |
      |                         long int
mulGlobal.h:124:8: note: in expansion of macro 'DUMPALLOCSIZ'
  124 |        DUMPALLOCSIZ;                                                        \
      |        ^~~~~~~~~~~~
mulSetup.c:1031:17: note: in expansion of macro 'CALLOC'
 1031 |   if(vects > 0) CALLOC(child->interList, vects, cube*, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:108:43: note: format string is defined here
  108 |                 "Total Memory Allocated: %d kilobytes (brk = 0x%x)\n", \
      |                                          ~^
      |                                           |
      |                                           int
      |                                          %ld
mulGlobal.h:128:23: warning: incompatible implicit declaration of built-in function 'exit' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch8;;]
  128 |        if(FLAG == ON) exit(0);                                              \
      |                       ^~~~
mulSetup.c:1031:17: note: in expansion of macro 'CALLOC'
 1031 |   if(vects > 0) CALLOC(child->interList, vects, cube*, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:128:23: note: include '<stdlib.h>' or provide a declaration of 'exit'
  128 |        if(FLAG == ON) exit(0);                                              \
      |                       ^~~~
mulSetup.c:1031:17: note: in expansion of macro 'CALLOC'
 1031 |   if(vects > 0) CALLOC(child->interList, vects, cube*, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:144:10: warning: incompatible implicit declaration of built-in function 'exit' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch8;;]
  144 |          exit(0);                                                           \
      |          ^~~~
mulSetup.c:1031:17: note: in expansion of macro 'CALLOC'
 1031 |   if(vects > 0) CALLOC(child->interList, vects, cube*, ON, AMSC);
      |                 ^~~~~~
mulGlobal.h:144:10: note: include '<stdlib.h>' or provide a declaration of 'exit'
  144 |          exit(0);                                                           \
      |          ^~~~
mulSetup.c:1031:17: note: in expansion of macro 'CALLOC'
 1031 |   if(vects > 0) CALLOC(child->interList, vects, cube*, ON, AMSC);
      |                 ^~~~~~
mulSetup.c: At top level:
mulSetup.c:1040:8: error: return type defaults to 'int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int8;;]
 1040 | static getAllInter(sys)
      |        ^~~~~~~~~~~
mulSetup.c:1062:8: error: return type defaults to 'int' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int8;;]
 1062 | static set_vector_masks(sys)
      |        ^~~~~~~~~~~~~~~~
make[1]: *** [<builtin>: mulSetup.o] Error 1
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:28: fastcap] Error 2

```

</details>

## Metadata

```
 nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.11, NixOS, 25.05 (Warbler), 25.05`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
```

## Notify maintainers

@fedeinthemix 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fa/fastcap/package.nix
+++ b/pkgs/by-name/fa/fastcap/package.nix
@@ -54,6 +54,9 @@ stdenv.mkDerivation rec {
     "all"
   ];
 
+  # GCC 14 makes these errors by default
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=implicit-function-declaration -Wno-error=return-mismatch -Wno-error=implicit-int";
+
   outputs = [
     "out"
     "doc"

